### PR TITLE
Add forklift to remaining jobs

### PR DESCRIPTION
**Pull request by AndWeHaveAPlan**
Add [forklift caching](https://gitlab.parity.io/parity/infrastructure/ci_cd/forklift/forklift) to remainig jobs

by .sh and .py scripts:
- cargo-check-each-crate x6 (`.gitlab/check-each-crate.py`)
- build-linux-stable (`polkadot/scripts/build-only-wasm.sh`)

by before_script:
- build-linux-substrate
- build-subkey-linux (with `.build-subkey` job)
- cargo-check-benches x2

**To disable feature set FORKLIFT_BYPASS variable to true in [project settings in gitlab](https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/settings/ci_cd)**
(forklift now handles FORKLIFT_BYPASS by itself)

### CODE REVIEW

File: .gitlab/check-each-crate.py [hunk 1]
```diff
@@ -55,7 +55,7 @@
 
 	print(f"Checking {crates[crate][0]}", file=sys.stderr)
 
-	res = subprocess.run(["cargo", "check", "--locked"], cwd = crates[crate][1])
+	res = subprocess.run(["forklift", "cargo", "check", "--locked"], cwd = crates[crate][1])
```
Comment (ggwpez) [inline .gitlab/check-each-crate.py:58]: Does the forklift have a `--quiet` option or something?  
I dont know if the noise here comes from forklift maybe https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5143188 but its difficult to see why the job failed.

Comment (AndWeHaveAPlan) [reply .gitlab/check-each-crate.py:58]: errors start at line [3708](https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5143188#L3708) with ```error[E0599]: the function or associated item `new` exists for struct `BenchmarkingState<Block<ExtrinsicWrapper<u64>>>`, but its trait bounds were not satisfied```

but yeah, that output seems redundant now, I'll fix it

FYI there are no flags as settings provided by config file. But it is possible to set vars by environment
`FORKLIFT_general_logLevel="warn"`

Comment (AndWeHaveAPlan) [reply .gitlab/check-each-crate.py:58]: fixed, from now on forklift will only write simple startup message and status report at the end (and default logLevel is now `warn`)

**Review by rcny — APPROVED** [review]

**Review by alvicsam — APPROVED** [review]

**Review by ggwpez — APPROVED** [review]
Approving to unblock.

File: polkadot/scripts/build-only-wasm.sh [hunk 2]
```diff
@@ -22,8 +22,8 @@ fi
 if [ -d $WASM_BUILDER_RUNNER ]; then
   export DEBUG=false
   export OUT_DIR="$PROJECT_ROOT/target/release/build"
-  cargo run --release --manifest-path="$WASM_BUILDER_RUNNER/Cargo.toml" \
+  forklift cargo run --release --manifest-path="$WASM_BUILDER_RUNNER/Cargo.toml" \
     | grep -vE "cargo:rerun-if-|Executing build command"
 else
-  cargo build --release -p $1
+  forklift cargo build --release -p $1
```
Comment (ggwpez) [inline polkadot/scripts/build-only-wasm.sh:28]: Actually i think this script is meant for endusers - not for CI.  
Can you maybe run it through the `RUSTC_WRAPPER` env variable like `RUSTC_WRAPPER=forklift ./build-only-wasm.sh`?

Comment (AndWeHaveAPlan) [reply polkadot/scripts/build-only-wasm.sh]: Makes sense. But wrappers need a "coordinator"  #process, and I don't like the idea of copy/paste script just for ci, so I added check for forklift

**Review by ggwpez — APPROVED** [review]

File: polkadot/scripts/build-only-wasm.sh (hunk 2 above)
Comment (ggwpez) [reply polkadot/scripts/build-only-wasm.sh]: Yup sounds good.

**Pull request merged**